airflow/www_rbac/views.py

```diff
@@ -204,10 +204,10 @@ def get_int_arg(value, default=0):
                                            search=arg_search_query,
                                            showPaused=not hide_paused),
             dag_ids_in_page=page_dag_ids,
-            auto_complete_data=auto_complete_data,
-            view_only=is_view_only(g.user, self.appbuilder))
+            auto_complete_data=auto_complete_data)
 
     @expose('/dag_stats')
+    @has_dag_access(can_dag_read=True)
     @has_access
     @provide_session
     def dag_stats(self, session=None):
```

Comment:
This isn't a per-DAG view, meaning you'll have to filter the query conditionally to limit the DAG to only ones that he user has access to. Somehow the query needs to be altered:
```python
dag_ids = sm.get_accessible_dag_ids():
if dag_ids:
    qry = qry.filter(dag_id.in_(dag_ids))
```
Or something like this.